src/main/java/net/whitbeck/rdbparser/ListpackList.java

```diff
@@ -0,0 +1,170 @@
+package net.whitbeck.rdbparser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import java.nio.charset.Charset;
+
+class ListpackList extends LazyList<byte[]> {
+    private static final Charset ASCII = Charset.forName("ASCII");
+
+    // Taken from
+    // https://github.com/redis/redis/blob/7.0.11/src/listpack.c#L55-L95C4
+    private static final int LP_ENCODING_7BIT_UINT = 0;
+    private static final int LP_ENCODING_7BIT_UINT_MASK = 0x80;
+    private static final int LP_ENCODING_6BIT_STR = 0x80;
+    private static final int LP_ENCODING_6BIT_STR_MASK = 0xC0;
+    private static final int LP_ENCODING_13BIT_INT = 0xC0;
+    private static final int LP_ENCODING_13BIT_INT_MASK = 0xE0;
+    private static final int LP_ENCODING_12BIT_STR = 0xE0;
+    private static final int LP_ENCODING_12BIT_STR_MASK = 0xF0;
+
+    // Sub encodings
+    private static final int LP_ENCODING_16BIT_INT = 0xF1;
+    private static final int LP_ENCODING_16BIT_INT_MASK = 0xFF;
+    private static final int LP_ENCODING_24BIT_INT = 0xF2;
+    private static final int LP_ENCODING_24BIT_INT_MASK = 0xFF;
+    private static final int LP_ENCODING_32BIT_INT = 0xF3;
+    private static final int LP_ENCODING_32BIT_INT_MASK = 0xFF;
+    private static final int LP_ENCODING_64BIT_INT = 0xF4;
+    private static final int LP_ENCODING_64BIT_INT_MASK = 0xFF;
+    private static final int LP_ENCODING_32BIT_STR = 0xF0;
+    private static final int LP_ENCODING_32BIT_STR_MASK = 0xFF;
+
+    private final byte[] envelope;
+
+    ListpackList(byte[] envelope) {
+        this.envelope = envelope;
+    }
+
+    class ListpackParser {
```

Comment:
```suggestion
    private class ListpackParser {
```

This seems like an implementation detail that we wouldn't want to reference outside of the ListpackList class.